ui/src/main/java/edu/wpi/grip/ui/pipeline/PipelineController.java

```diff
@@ -278,13 +273,13 @@ public void onStepMoved(StepMovedEvent event) {
     @Subscribe
     public void onConnectionAdded(ConnectionAddedEvent event) {
         // Add the new connection view
-        platform.runAsSoonAsPossible(() -> addConnectionView(event.getConnection()));
+        addConnectionView(event.getConnection());
     }
 
     @Subscribe
     public void onConnectionRemoved(ConnectionRemovedEvent event) {
         // Remove the control that corresponds with the connection that was removed
-        platform.runAsSoonAsPossible(() -> {
+        PlatformImpl.runAndWait(() -> {
```

Comment:
Okay, can you leave comments on the lines saying that this is intentional, and that it won't break anything. And also include that there should be no events posted inside of the blocks of code.